sdk/core/core/src/az_span.c

```diff
@@ -153,6 +154,30 @@ AZ_NODISCARD az_result az_span_atou32(az_span span, uint32_t* out_number)
   return AZ_OK;
 }
 
+AZ_NODISCARD az_result az_span_atod(az_span span, az_span* conversion_buffer, double* out_number)
```

Comment:
The longest decimal integer number that can be parsed as 64-bit iee754 floating point is `179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368`
, t is 309 characters long, plus one character for size. can we afford to allocate 310 bytes on stack internally without requirngg to pass conversion buffer to us?
This is the "worst case" scenario, I know that our curent implementation probably won't handle anything biffer than 2^52 (`4503599627370496`, 16 characters long + sign = 17)

Reply:
Ah sorry, I missed the tests - it looks like we CAN parse decimal points now, which is great, but my numbers from above do not make sense. But the principle of the question stays - there must be another limited size buffer that can fit any possible decimal number that `double` can represent, I can try to find it out.

Reply:
In .NET that number was 769, which we stackalloc'd:
https://source.dot.net/#System.Private.CoreLib/Number.NumberBuffer.cs,15
https://source.dot.net/#System.Private.CoreLib/Utf8Parser.Float.cs,67

Reply:
Btw, there are some really extensive/useful test cases here:
https://github.com/dotnet/runtime/blob/6b6d9135298500a753b3393b1c7cc4aa94617aa2/src/libraries/Common/tests/System/RealParserTestsBase.netcoreapp.cs#L140-L355

Reply:
Thank you Ahson - I think this is the number I was looking to find!

Reply:
We might not that lucky to be able to allocate that much stack into embebed devices right?
That's why I believe it is better to delegate that responsibility to customers.

And yes @antkmsft  atof() handles decimal point. However, out current implementation of dtoa() does not. So we still have some issue over there to fix that.

I believe we provide more flexibility by letting customers decide how big is the buffer they need for the conversion.

Reply:
I agree - across small devices, there seem to be 2 kinds for us: 1. embedded devices that do not have HTTP stack (for IoT SDK) - they may have 4-8KB memory, then 769 bytes is a luxury.
2. Devices with 32MB memory that use HTTP stack, they can handle 769 bytes.

We can help some customers who know that they have enough memory, and just want to be sure they can parse any possible value, without wasting extra memory at the same time.
```
enum
{
    // Conversion buffer size that should be able to fit any possible decimal value that can fit into double precision IEEE754
    AZ_SPAN_ATOD_SUFFICIENT_CONVERSION_BUFFER = 769,
}
```

...maybe. It is up to you to take this suggestion.

Reply:
Or maybe in the doxygen we can mention that conversion buffer of 769 bytes is always sufficient.